### PR TITLE
Updated migration

### PR DESCRIPTION
**Pull request by lucassaldanha**

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    LOG.trace("Migrating private storage database...");
+
+    final long chainHeadBlockNumber = blockchain.getChainHeadBlockNumber();
+
+    for (int blockNumber = 0; blockNumber <= chainHeadBlockNumber; blockNumber++) {
+      final Block block = blockchain.getBlockByNumber(blockNumber).orElseThrow();
+      final Hash blockHash = block.getHash();
+      final BlockHeader blockHeader = block.getHeader();
+      LOG.trace(
+          "Processing block {} ({}/{})", blockHash, blockNumber + 1, chainHeadBlockNumber + 1);
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java]: I think the blockNumber + 1 and chainHeadBlockNumber +1 could be confusing ...

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+      final PrivacyGroupHeadBlockMap privacyGroupHeadBlockMap =
+          createPrivacyGroupHeadBlockMap(blockHeader);
+
+      final List<Transaction> pmtsInBlock = findPMTsInBlock(block);
+      for (int pmtIndex = 0; pmtIndex < pmtsInBlock.size(); pmtIndex++) {
+        final Transaction pmt = pmtsInBlock.get(pmtIndex);
+        LOG.trace("Processing PMT {} ({}/{})", pmt.getHash(), pmtIndex + 1, pmtsInBlock.size());
+
+        final ReceiveResponse receiveResponse =
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java]: In real life you might not get a PTx for every PMT ... If that is the case you just start processing the next PMT :-)

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+  private PrivacyGroupHeadBlockMap createPrivacyGroupHeadBlockMap(final BlockHeader blockHeader) {
+    final PrivacyGroupHeadBlockMap privacyGroupHeadBlockHash =
+        new PrivacyGroupHeadBlockMap(
+            privateStateKeyValueStorage
+                .getPrivacyGroupHeadBlockMap(blockHeader.getParentHash())
+                .orElse(PrivacyGroupHeadBlockMap.EMPTY));
+    privateStateKeyValueStorage
+        .updater()
+        .putPrivacyGroupHeadBlockMap(blockHeader.getHash(), privacyGroupHeadBlockHash)
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java]: There is no need to store an empty map!  You could store this map once all PMTs have been executed, but it should only happen if the map is not empty!

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      final BlockHeader blockHeader,
+      final String privacyGroupId,
+      final PrivateTransaction privateTransaction) {
+    final PrivateTransactionSimulatorResult simulatorResult =
+        privateTransactionSimulator
+            .process(privacyGroupId, enclaveKey.toBase64String(), privateTransaction, blockHeader)
+            .orElseThrow();
+
+    LOG.trace(
+        "Executed private transaction {} (result = {}}",
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java]: s/Executed/Simulated/

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+            .getPrivateBlockMetadata(blockHeader.getHash(), privacyGroupIdBytes)
+            .orElseGet(PrivateBlockMetadata::empty);
+
+    final PrivateTransactionMetadata privateTxMetadata =
+        new PrivateTransactionMetadata(
+            privacyMarkerTxHash, txResult.getResultingRootHash().orElseThrow());
+    privateBlockMetadata.addPrivateTransactionMetadata(privateTxMetadata);
+
+    updater.putPrivateBlockMetadata(
+        Bytes32.wrap(blockHeader.getHash()), privacyGroupIdBytes, privateBlockMetadata);
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java]: Do you really have to wrap that? I thought that a hash is a Bytes32 already ...

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java (hunk 2 above)
Comment (iikirilov) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/migration/PrivateStorageMetadataMigration.java]: Yes, if it is not a privacy group you are in you will not have PTx for that PMT

**Pull request closed without merging**